machinery-helper/Rakefile

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) 2013-2015 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of version 3 of the GNU General Public License as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE about this file by physical or electronic mail,
+# you may find current contact information at www.suse.com
+
+task :default => "check_git_head"
+
+dir = File.expand_path(File.dirname(__FILE__))
+
+go_files = Dir.glob(File.join(dir, "*.go"))
+
+task :check_git_head do
+  git_head_file = File.join(dir, ".git_head")
+  current_head = `git rev-parse HEAD`.chomp if system("which git > /dev/null 2>&1")
+  old_head = File.read(git_head_file) if File.exist?(git_head_file)
+  if current_head != old_head
+    File.write(git_head_file, current_head)
+    Rake::Task["build_helper"].invoke
+  else
+    Rake::Task["machinery-helper"].invoke
+  end
+end
+
+file "machinery-helper" => go_files do |t|
+  Rake::Task["build_helper"].invoke
+end
+
+desc "Build machinery-helper with go"
+task :build_helper do
+  if system("which go > /dev/null 2>&1")
+    arch = `uname -p`.chomp
+    if ["x86_64"].include?(arch)
+      Dir.chdir(dir) do
+        puts("Building machinery-helper binary.")
+        system("go build")
+      end
+    else
+      puts("The hardware architecture #{arch} is not yet supported by the machinery-helper.")
```

Comment:
This and the other message should be recognizable as warnings and be put out on STDERR.